apps/scale-cluster/src/scale_cluster.py

```diff
@@ -33,33 +33,37 @@ def get_ec2_spending_month_to_date():
     return float(response['ResultsByTime'][0]['Total']['UnblendedCost']['Amount'])
 
 
-def update_compute_cluster_size(max_vcpus):
-    compute_environment_arn = environ['COMPUTE_ENVIRONMENT_ARN']
+def set_max_vcpus(compute_environment_arn, max_vcpus):
     print(f'Updating {compute_environment_arn} maxvCpus to {max_vcpus}')
     BATCH.update_compute_environment(
         computeEnvironment=compute_environment_arn,
         computeResources={'maxvCpus': max_vcpus},
     )
 
 
-def get_max_vcpus(today, budget, spending, default_max_vcpus, expanded_max_vcpus, required_surplus):
+def get_max_vcpus(today, monthly_compute_budget, month_to_date_compute_spending, default_max_vcpus, expanded_max_vcpus,
+                  required_surplus):
     days_in_month = calendar.monthrange(today.year, today.month)[1]
+    month_to_date_compute_budget = monthly_compute_budget * today.day / days_in_month
+    available_surplus = month_to_date_compute_budget - month_to_date_compute_spending
 
-    target_spending = (budget / days_in_month) * today.day
-    max_spending_for_next_day = (target_spending - spending)
+    print(f'Month-to-date compute budget: ${month_to_date_compute_budget:,.2f}')
+    print(f'Month-to-date compute spending: ${month_to_date_compute_spending:,.2f}')
+    print(f'Available surplus: ${available_surplus:,.2f}')
+    print(f'Required surplus: ${required_surplus:,.2f}')
 
-    if max_spending_for_next_day >= required_surplus:
-        max_vcpus = expanded_max_vcpus
-    else:
+    if available_surplus < required_surplus:
```

Comment:
With a binary switch between fleet sizes, I would almost rather return this (fn would be `use_expand_fleet` or something then), and keep the if-else logic in `lambda_handler` as it would drop two parameters from this function that are only used in the if-else. 

If we go to a more dynamic fleet size, I would want it so you get something between a zero and one back, and use that value to set the vcpus. Something like:
```python
fleet_size = default_max_vcpus + epansion_factor * (absolute_max_vcpus - default_max_vcpus)
```

Reply:
"Default", "Expanded" and "Surplus" are the words I'd like to make more expressive, I'm open to suggestions to replace any/all of them.